modules/SofaConstraint/SlidingConstraint.inl

```diff
@@ -162,30 +160,33 @@ void SlidingConstraint<DataTypes>::getConstraintResolution(const ConstraintParam
 template<class DataTypes>
 void SlidingConstraint<DataTypes>::draw(const core::visual::VisualParams* vparams)
 {
-#ifndef SOFA_NO_OPENGL
     if (!vparams->displayFlags().getShowInteractionForceFields()) return;
 
-    glDisable(GL_LIGHTING);
-    glPointSize(10);
-    glBegin(GL_POINTS);
+    vparams->drawTool()->saveLastState();
+
+    vparams->drawTool()->disableLighting();
+
+    sofa::defaulttype::RGBAColor color;
+
     if(thirdConstraint<0)
-        glColor4f(1,1,0,1);
+        color = sofa::defaulttype::RGBAColor::yellow();
     else if(thirdConstraint>0)
-        glColor4f(0,1,0,1);
+        color = sofa::defaulttype::RGBAColor::green();
     else
-        glColor4f(1,0,1,1);
-    helper::gl::glVertexT((this->mstate1->read(core::ConstVecCoordId::position())->getValue())[m1.getValue()]);
-    //      helper::gl::glVertexT((*this->object2->read(sofa::core::ConstVecCoordId::position())->getValue())[m3]);
-    //      helper::gl::glVertexT(proj);
-    glEnd();
-
-    glBegin(GL_LINES);
-    glColor4f(0,0,1,1);
-    helper::gl::glVertexT((this->mstate2->read(core::ConstVecCoordId::position())->getValue())[m2a.getValue()]);
-    helper::gl::glVertexT((this->mstate2->read(core::ConstVecCoordId::position())->getValue())[m2b.getValue()]);
-    glEnd();
-    glPointSize(1);
-#endif /* SOFA_NO_OPENGL */
+        color = sofa::defaulttype::RGBAColor::magenta();
```

Comment:
Same for the following block of code. it would be simpler with a `do_whatever(...)` lambda function